scripts/imgtool/image.py

```diff
@@ -69,8 +73,17 @@ class Image():
     @classmethod
     def load(cls, path, included_header=False, **kwargs):
         """Load an image from a given file"""
-        with open(path, 'rb') as f:
-            payload = f.read()
+        ext = os.path.splitext(path)[1][1:].lower()
```

Comment:
This works fine if the filename has no extension, since `ext` will be an empty string

Reply:
It does rely on [1:] in python not failing on an empty string (which was surprising to me).

Reply:
This is legit:

```
The slice of s from i to j is defined as the sequence of items with index k such that i <= k < j. If i or j is greater than len(s), use len(s).
```

https://docs.python.org/3/library/stdtypes.html#typesseq-common